packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -61,7 +61,7 @@ export interface Store {
   channelUpdatedFeed(channelId: string): Observable<ChannelStoreEntry>;
 
   getAddress(): string;
-  signAndAddState(channelId: string, stateVars: StateVariables);
+  signAndAddState(channelId: string, stateVars: StateVariables): void;
```

Comment:
```suggestion
  signAndAddState(channelId: string, stateVars: StateVariables): Promise<void>;
```

(Adding a state in general is async.)